packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -477,25 +494,29 @@ class _CupertinoDatePickerDateTimeState extends State<CupertinoDatePicker> {
         if (widget.use24hFormat) {
           selectedHour = index;
           widget.onDateTimeChanged(_getDateTime());
-        }
-        else {
-          final int currentHourIn24h = selectedHour + selectedAmPm * 12;
+        } else {
+          selectedHour = index % 12;
```

Comment:
Do we really need a new `previousHourIndex`? It seems like selectedHour is only read while building onDateTimeChanged(_getDateTime()) which is deferred in this branch. Can we just compare index against selectedHour and set selectedHour after the animateToItem?

I may have missed something too.

Reply:
If we're in 12-hour mode, then selectedHour only ranges from 0-11 - so it won't work for us to compare against index which ranges from 0-23.

Reply:
Ah I see. Can you make sure this is commented at the definition of previousHourIndex or make it clearer by its name such as _previous24RangeHourIndex or some such to distinguish its public-facing-ness and range from `selectedHour`?

Reply:
Improved the docs in my latest commit